pkg/client/client.go

```diff
@@ -43,6 +48,42 @@ func New(cfg Config) (*CortexClient, error) {
 		"id":      cfg.ID,
 	}).Debugln("New ruler client created")
 
+	//if TLS flags, setup http client
+	if cfg.TLScaFile != "" && cfg.TLScertFile != "" && cfg.TLSkeyFile != "" {
+
+		// Load client cert
+		cert, err := tls.LoadX509KeyPair(cfg.TLScertFile, cfg.TLSkeyFile)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"tls-cert": cfg.TLScertFile,
+				"tls-key":  cfg.TLSkeyFile,
+			}).Fatal(err)
```

Comment:
Following with the return signature of this method `(*CortexClient, error)` instead of using `log.Fatal()` here I'd let the caller handle the error, i.e.:
```golang
log.WithError(err).WithFields(log.Fields{
	"tls-cert":  cfg.TLSCertFile,
	"tls-key":  cfg.TLSKeyFile,
}).Errorf("error loading tls files")
return nil, fmt.Errorf("client initialization unsuccessful")
```  

This way you get both things, all the necessary information for debugging and give control to the caller.